src/github/membership/index.ts

```diff
@@ -0,0 +1,151 @@
+  org?: string;
+  users?: string[];
+  path?: string;
+  token?: string;
+  spinner?: Ora;
+}
+
+type RETVAL = {
+  success: string[];
+  failure: string[];
+};
+
+/**
+ * Execute the `github-membership` command.
+ *
+ * @param {string} [args.org] GitHub organization name.
+ * @param {string[]} [args.users] GitHub user names to remove from membership.
+ * @param {string} [args.path] Path to a git directory.
+ * @param {string} [args.token] GitHub personal access token.
+ * @param {string} [args.spinner] Terminal spinner.
+ *
+ * @returns {Promise<object>} The success and failure results of the membership command.
+ */
+export const execute = async (args: IGitHubMembershipArgs): Promise<RETVAL> => {
+  const retVal: RETVAL = { success: [], failure: [] };
+
+  try {
+    const auth = args.token || (await getToken(args.spinner));
+    const github = new Octokit({ auth });
+    const org = (args.org || (await getRepository(args.path, args.spinner))?.owner)!;
+
+    if (args.users) {
+      const users = Array.isArray(args.users) ? args.users : [args.users];
+      const requests = [];
+
+      for (const user of users) {
+        /* https://octokit.github.io/rest.js/v21/#orgs-remove-member */
+        const request = github.orgs.removeMember({ org, username: user });
+
+        requests.push(request);
+      }
+
+      await Promise.allSettled(requests).then(results => {
+        results.forEach(result => {
+          if (result.status === 'fulfilled') {
+            const user = result.value.url.split('/').pop();
+
+            retVal.success.push(user!);
+          } else {
+            const user = result.reason.response.url.split('/').pop();
+
+            if (typeof user === 'string') {
+              retVal.failure.push(user);
+            }
+
+            handleErrorMessage(result.reason, 'github-membership', args.spinner);
+          }
+        });
+      });
+    } else {
+      /* https://octokit.github.io/rest.js/v21/#orgs-list-members */
+      const members = await github.paginate(github.orgs.listMembers, { org, per_page: 100 });
+
+      retVal.success = members.map(member => member.login).sort();
+    }
+  } catch (error: unknown) {
+    handleErrorMessage(error, 'github-membership', args.spinner);
+
+    throw error;
+  }
+
+  return retVal;
+};
+
+export default (spinner: Ora): Command => {
+  const command = new Command('github-membership');
+
+  return command
+    .description('manage GitHub organization membership')
+    .option('-o, --org [org]', 'GitHub organization name; defaults to repository owner')
+    .option('-d --delete <users...>', 'remove members')
+    .option('-l --list', 'list members')
+    .option('-p, --path <path>', 'git directory')
+    .option('-t, --token <token>', 'access token')
+    .action(async () => {
+      try {
+        spinner.start();
+
+        const options = command.opts();
+        const results = await execute({
+          org: options.org,
+          users: options.delete,
```

Comment:
About this [comment](https://github.com/zendeskgarden/scripts/pull/314/files#r2512178528). 

I'm a little confused. If `users` only accepts `string[] | undefined`, TS would throw - unless `options.delete` is of type `any`. Is that the case here?

Reply:
All commander.js options are of type `any`. The code we're swirling around here is simply a belt+suspenders mechanism to ensure `users` is being treated as the expected array and not turned into `[j, z, e, m, p, e, l]` for the limited corner case where the script would be used to delete one user. The intended & expected usage is bulk deletion. To say it another way:

- CLI could naturally be used to delete one or more users (and the script ensures type safety)
- API woud naturally be used to delete users in bulk